### PR TITLE
Cancel booster indication (EXPOSUREAPP-10232)

### PR DESCRIPTION
**Pull request by mtwalli**
- Booster indication is getting cancelled automatically if user is not eligible for booster anymore
- Booster rules runs 
  - after scanning new certificates
  - Recycle certificate
  - Restore certificate

### Testing 
- Scan certificates set from the attached one to the ticket where user is eligible (1,2)
- Scan the booster certificate (3) -> the booster indication message should be gone after the scan

## Ticket
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-10232

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/qrcode/handler/DccQrCodeHandler.kt [hunk 1]
```diff
@@ -17,21 +18,23 @@ class DccQrCodeHandler @Inject constructor(
...
+    suspend fun handleQrCode(dccQrCode: DccQrCode): CertificateContainerId {
+        dscSignatureValidator.validateSignature(dccData = dccQrCode.data)
+        return when (dccQrCode) {
+            is RecoveryCertificateQRCode -> recoveryCertificateRepository.registerCertificate(dccQrCode).containerId
+            is VaccinationCertificateQRCode -> vaccinationRepository.registerCertificate(dccQrCode).containerId
+            is TestCertificateQRCode -> testCertificateRepository.registerCertificate(dccQrCode).containerId
+            else -> throw UnsupportedQrCodeException()
+        }.also {
+            boosterCheckScheduler.scheduleNow()
         }
```
Comment (chiljamgossow) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/qrcode/handler/DccQrCodeHandler.kt:38]: Is it possible to move booster check into registerCertificate() so the workaround with the 3000 ms delay is not needed?

**Comment by KathrinLuecke** [comment]
I am not sure why UA should review this as no text changes are included?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/qrcode/handler/DccQrCodeHandler.kt (hunk 1 above)
Comment (mtwalli) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/qrcode/handler/DccQrCodeHandler.kt:38]: Unfortunately it is not possible for the following reason:
- it creates a cyclic DI dependencies between BoosterNotificationService  <-> VaccinationRepository

**Comment by mtwalli** [comment]
> I am not sure why UA should review this as no text changes are included?

No UA is required here

**Review by CV113 — DISMISSED** [review]
UA approved

**Review by SamuraiKek — APPROVED** [review]
Tested with all certs from the ticket. Works nicely.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/qrcode/handler/DccQrCodeHandler.kt (hunk 1 above)
Comment (chiljamgossow) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/qrcode/handler/DccQrCodeHandler.kt:38]: 🤷‍♀️

**Review by chiljamgossow — APPROVED** [review]

**Pull request merged**